src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -815,6 +816,10 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
                     pattern = rList[0];
                     if (rList.Count > 1)
                     {
+                        if(rList[1] is ScriptBlock)
```

Comment:
Shouldn't this also support a `MatchEvaluator` delegate? Do that we can use a method from a PowerShell or .NET class.
That is, in addition to the scriptblock.

Reply:
Good call! 
Wondering whether it should accept/evaluate `System.Delegate`, `System.Func<string>` and `System.Func<Match, string>` as well...

Reply:
Nevermind, cast will fail.

Reply:
@powercode Passing a `PSMethodInfo` object would make sense as a use case, but I'm not sure how to safely construct a scriptblock or delegate that dispatches it.

Any ideas?